src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -242,7 +242,8 @@ CHIP_ERROR Hash_SHA256_stream::GetDigest(MutableByteSpan & out_buffer)
     mbedtls_sha256_context * context = to_inner_hash_sha256_context(&mContext);
 
     // Back-up context as we are about to finalize the hash to extract digest.
-    mbedtls_sha256_context previous_ctx = *context;
+    mbedtls_sha256_context previous_ctx;
+    mbedtls_sha256_clone(&previous_ctx, context);
```

Comment:
Does this rely on special behavior of mbedtls_sha256_clone that would come in a later ESP-IDF update? What is different here?

Reply:
My `tools/esp-idf/components/mbedtls/port/esp32/esp_sha256.c` (unlike the other two copies of mbedtls in tools/esp-idf) has:
```
    *dst = *src;

    if (src->mode == ESP_MBEDTLS_SHA256_HARDWARE) {
        /* Copy hardware digest state out to cloned state,
           which will become a software digest.
        */
        esp_sha_read_digest_state(SHA2_256, dst->state);
        dst->mode = ESP_MBEDTLS_SHA256_SOFTWARE;
    }
```
so that is presumably the key here.